velox/common/memory/SharedArbitrator.cpp

```diff
@@ -467,7 +467,7 @@ void SharedArbitrator::addPool(const std::shared_ptr<MemoryPool>& pool) {
 }
 
 void SharedArbitrator::removePool(MemoryPool* pool) {
-  VELOX_CHECK_EQ(pool->reservedBytes(), 0);
+  VELOX_DCHECK_EQ(pool->reservedBytes(), 0);
```

Comment:
We have to crash the server at this point as this is capacity leak. We need to fix instead of ignoring it. Thanks!